provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -276,11 +276,17 @@ func generateSchema() schema.PackageSpec {
 							"Also consider setting `nodeAssociatePublicIpAddress: false` for fully private workers.",
 					},
 					"nodeGroupOptions": {
-						TypeSpec:    schema.TypeSpec{Ref: "#/types/eks:index:ClusterNodeGroupOptions"},
+						TypeSpec: schema.TypeSpec{
+							Ref:   "#/types/eks:index:ClusterNodeGroupOptions",
+							Plain: true,
+						},
 						Description: "The common configuration settings for NodeGroups.",
 					},
 					"nodeAssociatePublicIpAddress": {
-						TypeSpec: schema.TypeSpec{Type: "boolean"},
+						TypeSpec: schema.TypeSpec{
+							Type:  "boolean",
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L963-L969